### PR TITLE
feat(TextInput): Added OUIA for TextInput

### PR DESCRIPTION
**Pull request by quarckster**
Part of #2425

What: Added OUIA props to TextInput component

### CODE REVIEW

**Comment by quarckster** [comment]
@redallen @karelhala test failed:
```
TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    packages/patternfly-4/react-core/src/components/TextInput/TextInput.tsx:114:50 - error TS2345: Argument of type 'typeof TextInput' is not assignable to parameter of type 'ComponentClass<TextInputProps & InjectedOuiaProps, any> | FunctionComponent<TextInputProps & InjectedOuiaProps>'.
      Type 'typeof TextInput' is not assignable to type 'ComponentClass<TextInputProps & InjectedOuiaProps, any>'.
        Types of property 'defaultProps' are incompatible.
          Type '{ 'aria-label': string; className: string; isRequired: boolean; isValid: boolean; isDisabled: boolean; isReadOnly: boolean; type: string; onChange: () => any; }' is not assignable to type 'Partial<TextInputProps & InjectedOuiaProps>'.
            Types of property 'type' are incompatible.
              Type 'string' is not assignable to type '"number" | "date" | "time" | "text" | "datetime-local" | "email" | "month" | "password" | "search" | "tel" | "url"'.

    114 const TextInputWithOuiaContext = withOuiaContext(TextInput)
```

**Comment by vmuzikar** [comment]
@quarckster Hi, I just wanted to ask what's the status on this please. Any ETA? :)

**Comment by quarckster** [comment]
@vmuzikar I wait for help from @redallen or @karelhala. My typescript knowledge is limited.

**Comment by quarckster** [comment]
@redallen @tlabaj  tests passed

File: packages/patternfly-4/react-core/src/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -114,10 +117,14 @@ class TextInputBase extends React.Component<TextInputProps> {
         disabled={isDisabled}
         readOnly={isReadOnly}
         ref={innerRef}
+        {...(ouiaContext.isOuia && {
+          'data-ouia-component-type': 'PF4/TextInput',
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/TextInput/TextInput.tsx]: The rest of the OUIA enabled components do not include the `PF4/` prefix. Why are you adding it for only TextInput?

Comment (quarckster) [reply packages/patternfly-4/react-core/src/components/TextInput/TextInput.tsx]: @redallen Actually I created a PR with changes for ouia enabled components https://github.com/patternfly/patternfly-react/pull/3622

Comment (redallen) [reply packages/patternfly-4/react-core/src/components/TextInput/TextInput.tsx]: Very well. You'll have to handle not having the `PF4/` prefix for older versions in your QA suite.

**Comment by redallen** [comment]
Can't merge since there are conflicts.

**Comment by quarckster** [comment]
@redallen fixed

**Comment by redallen** [comment]
Should be rebased on our `v4` branch now that #3927 is in. Closing.

**Pull request closed without merging**